Add bbe for RegExp type (#4193)
ballerina-platform/ballerina-distribution: examples/regexp-type/regexp_type.md

# RegExp 

`RegExp` is a built-in subtype of `anydata`.

> The lang.regexp module contains various regular expression related functions. You can learn more about these functions and how to use them by referring to the lang.regexp API documentation.

Reviewer: ```suggestion
The `lang.regexp` module contains various functions related to regular expressions. You can learn more about these functions and how to use them by referring to the `lang.regexp` API documentation.
```
Should we add the API Docs link?

Author: We can add it in the Related Links section. Ryt?

> 
> The `RegExp` type is defined in the `lang.regexp` module, and can also be referenced using the type alias in the `lang.string` module. The RegExp type conforms to a subset of the ECMAScript specification for regular expressions.

Reviewer 2: ```suggestion
The `RegExp` type is defined in the `lang.regexp` module, and can also be referenced using the type alias named the same in the `lang.string` module. The `RegExp` type conforms to a subset of the ECMAScript specification for regular expressions.
```

> Users can declare `RegExp` variables in two ways. Using the `string`as module prefix or `regexp` langlib as the module prefix. In the second case users have explicitly import the `regexp` module.

Reviewer 3: ```suggestion
Users can declare `RegExp` variables in two ways. The first case is to use the `string`as module prefix or `regexp` langlib as the module prefix. In the second case users have explicitly import the `regexp` module.
```

@praneesha wdyt?

> Users can declare `RegExp` variables in two ways. Using the `string`as module prefix or `regexp` langlib as the module prefix. In the second case users have explicitly import the `regexp` module.

Reviewer: ```suggestion
You can declare `RegExp` variables in two ways either by using the `string` as the module prefix or by using the `regexp` langlib as the module prefix. In the second method, the `regexp` module needs to be imported explicitly.
```

::: code regexp_type.bal :::

::: out regexp_type.out :::

## Related links
- [Ballerina regular expression grammar](https://ballerina.io/spec/lang/master/#section_10.1)
- [RegExp langlib functions](/learn/by-example/regexp-operations)

> - [RegExp API Docs](https://lib.ballerina.io/ballerina/lang.regexp)

Reviewer 2: Add string langlib
